core/mqtt/server.go

```diff
@@ -83,22 +88,24 @@ func (s *Server) Run(ctx context.Context) {
 		},
 		s.brokerOptions)
 	if err != nil {
-		panic(err)
+		s.LogErrorfAndExit("failed to create MQTT broker: %s", err.Error())
 	}
 
 	s.MQTTBroker = broker
 
-	go func() {
-		if err := broker.Start(); err != nil {
-			panic(err)
-		}
-	}()
+	if err := broker.Start(); err != nil {
```

Comment:
you sure this is non-blocking?

Reply:
yeah I tested it, works.
This was the old MQTT broker that was blocking I guess.